packages/opentelemetry-resources/src/ResourceImpl.ts

```diff
@@ -177,3 +193,61 @@ function guardedRawAttributes(
     return [k, v];
   });
 }
+
+function validateSchemaUrl(schemaUrl?: string): string | undefined {
+  if (schemaUrl === undefined || schemaUrl === '') {
+    return schemaUrl;
+  }
+
+  try {
+    const url = new URL(schemaUrl);
+
+    if (url.protocol !== 'http:' && url.protocol !== 'https:') {
+      diag.warn(
+        'Invalid schema URL format: "%s". Schema URL must be a valid URI with http:// or https://. Schema URL will be ignored.',
```

Comment:
nit: string used twice can be a constant

Reply:
addressed in [5190bcc](https://github.com/open-telemetry/opentelemetry-js/pull/5753/commits/5190bccf7ed1fc64e5961003cd177b2e29189279)